postgres/infoschema.go

```diff
@@ -336,6 +342,87 @@ func getConstraints(conv *internal.Conv, db *sql.DB, table schemaAndName) ([]str
 	return primaryKeys, m, nil
 }
 
+// getForeignKeys return list all the foreign keys constraints.
+func getForeignKeys(conv *internal.Conv, db *sql.DB, table schemaAndName) (foreignKeys []schema.ForeignKey, err error) {
+	refTables, err := getRefTables(db, table)
+	if err != nil {
+		return nil, err
+	}
+	for _, refTable := range refTables {
+		fkey, err := getForeignKey(conv, db, table, refTable)
+		if err != nil {
+			return nil, err
+		}
+		foreignKeys = append(foreignKeys, fkey)
+	}
+	return foreignKeys, nil
+}
+
+// getRefTables return the list of referenced tables for
+// the selected schema and table.
+func getRefTables(db *sql.DB, table schemaAndName) ([]schemaAndName, error) {
+	q := `SELECT c.TABLE_SCHEMA,c.TABLE_NAME 
+		FROM INFORMATION_SCHEMA.CONSTRAINT_TABLE_USAGE c
+		INNER JOIN INFORMATION_SCHEMA.TABLE_CONSTRAINTS t
+			ON c.CONSTRAINT_NAME = t.CONSTRAINT_NAME
+		WHERE constraint_type='FOREIGN KEY' 
+			AND t.table_schema=$1
+			AND t.table_name=$2`
+	rows, err := db.Query(q, table.schema, table.name)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't get reference tables: %w", err)
+	}
+	defer rows.Close()
+	var refTableSchema, refTableName string
+	var refTables []schemaAndName
+	for rows.Next() {
+		rows.Scan(&refTableSchema, &refTableName)
+		refTables = append(refTables, schemaAndName{schema: refTableSchema, name: refTableName})
+	}
+	return refTables, nil
+}
+
+// getForeignKey returns the foreign key constraint for
+// a particular referenced table in the selected table
+// and database.
+func getForeignKey(conv *internal.Conv, db *sql.DB, table schemaAndName, refTable schemaAndName) (schema.ForeignKey, error) {
+	q := `SELECT k.column_name,ref.column_name
+		FROM INFORMATION_SCHEMA.referential_constraints as r
```

Comment:
This query is pretty subtle.
Maybe a comment pointing to postgres documentation on this aspect of INFORMATION_SCHEMA?

Reply:
There is nothing straight forward in documentation explaining this. But maybe we can just put link for the table we have used in our query.